### PR TITLE
grpc_set_ssl_roots_override_callback for Python

### PR DESCRIPTION
**Pull request by nathanielmanistaatgoogle**
Not yet passing interop tests but ready to at least begin review.

Since this is my first Cython change @robertwb has graciously agreed to also review.

### CODE REVIEW

**Comment by nathanielmanistaatgoogle** [comment]
@soltanmm and @robertwb: after changing `cimport libc` to `from libc.string cimport memcpy` and `libc.memcpy` to `memcpy` this now passes `run_interop_tests.py --use_docker -lpython --cloud_to_prod` and I believe is ready for review.

**Comment by soltanmm** [comment]
Does the 'now' in

> this now passes

imply that it wasn't passing before?

If so, that's a little baffling to me (but lgtm in any case).

**Comment by nathanielmanistaatgoogle** [comment]
@soltanmm: yes. Note the `--cloud_to_prod` flag, which wasn't included in what we were running Friday morning.

File: src/python/grpcio/grpc/_cython/_cygrpc/security.pxd.pxi [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+cimport cpython
```
Comment (robertwb) [inline src/python/grpcio/grpc/_cython/_cygrpc/security.pxd.pxi]: Why are you cimporting cpython here?

File: src/python/grpcio/grpc/_cython/_cygrpc/security.pyx.pxi [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+cimport cpython
```
Comment (robertwb) [inline src/python/grpcio/grpc/_cython/_cygrpc/security.pyx.pxi]: same

File: src/python/grpcio/grpc/_cython/_cygrpc/security.pxd.pxi (hunk 1 above)
Comment (nathanielmanistaatgoogle) [inline src/python/grpcio/grpc/_cython/_cygrpc/security.pxd.pxi]: This file started life as a fork of credentials.pxd.pxi, so... if this cimport doesn't need to be here, I'm happy to take it out.

File: src/python/grpcio/grpc/_cython/_cygrpc/security.pyx.pxi (hunk 2 above)
Comment (nathanielmanistaatgoogle) [inline src/python/grpcio/grpc/_cython/_cygrpc/security.pyx.pxi]: Analogous reason here - this file started life as a fork of credentials.pyx.pxi. But if there's no reason for the cimport I'm happy to remove it.

**Comment by nathanielmanistaatgoogle** [comment]
I've now dropped the `cimport cpython` from my two new files and changed the `.gitignore` file to match the moved credentials location; please quickly rereview.

**Comment by soltanmm** [comment]
You found a `Cloak of LGTM (+1)`!

Would you like to equip it now? *(yes/no)*   blinking-cursor

**Comment by jtattermusch** [comment]
Does this work when you remove make install-certs  from the python interop docker image?

**Comment by nathanielmanistaatgoogle** [comment]
@jtattermusch: Yes! Would you like a follow-on pull request removing that line? I apologize that I forgot to include it in this change.

**Comment by jtattermusch** [comment]
Yes, pls send a PR. Thanks!

**Comment by nathanielmanistaatgoogle** [comment]
Sent #5168.

**Pull request merged**